### PR TITLE
feat: Create unit tests for slices in Redux in I3S Explorer

### PR DESCRIPTION
**Pull request by MaximKYZ**
@belom88 Please review

### CODE REVIEW

File: src/redux/slices/flattened-sublayers-slice.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,1851 @@
...
+      token: undefined,
+      type: TilesetType.I3S,
+      url: "https://testurl",
+    },
+  ];
+
+  store.dispatch(
+    getFlattenedSublayers({ tilesetsData, buildingExplorerOpened: false })
+  );
+  await sleep(100);
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.spec.tsx:213]:
```suggestion
  await store.dispatch(
    getFlattenedSublayers({ tilesetsData, buildingExplorerOpened: false })
  );
```

File: src/redux/slices/flattened-sublayers-slice.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,1851 @@
...
+
+  tilesetsData[0]["id"] = "test_id_left";
+  store.dispatch(
+    getFlattenedSublayers({
+      tilesetsData,
+      buildingExplorerOpened: false,
+      side: ComparisonSideMode.left,
+    })
+  );
+  await sleep(100);
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.spec.tsx:235]:
```suggestion
  await store.dispatch(
    getFlattenedSublayers({
      tilesetsData,
      buildingExplorerOpened: false,
      side: ComparisonSideMode.left,
    })
  );
```

File: src/redux/slices/flattened-sublayers-slice.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,1851 @@
...
+
+  tilesetsData[0]["id"] = "test_id_right";
+  store.dispatch(
+    getFlattenedSublayers({
+      tilesetsData,
+      buildingExplorerOpened: false,
+      side: ComparisonSideMode.right,
+    })
+  );
+  await sleep(100);
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.spec.tsx:257]: Ditto

File: src/redux/slices/flattened-sublayers-slice.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,1851 @@
...
+      token: undefined,
+      type: TilesetType.I3S,
+      url: "https://testurl",
+    },
+  ];
+
+  store.dispatch(
+    getFlattenedSublayers({ tilesetsData, buildingExplorerOpened: false })
+  );
+  await sleep(100);
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.spec.tsx:294]: Ditto

File: src/redux/slices/flattened-sublayers-slice.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,1851 @@
...
+      token: undefined,
+      type: TilesetType.I3S,
+      url: "https://testurl",
+    },
+  ];
+
+  store.dispatch(
+    getFlattenedSublayers({ tilesetsData, buildingExplorerOpened: true })
+  );
+  await sleep(100);
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.spec.tsx:323]: Ditto

File: src/redux/slices/flattened-sublayers-slice.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,1851 @@
...
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.spec.tsx:352]: Ditto

File: src/types.ts [hunk 7]
```diff
@@ -358,7 +358,7 @@ export type TileSelectedColor = {
 export type TilesetMetadata = {
   id: string;
   url: string;
-  token: string;
+  token: string | undefined;
```
[the remark this answers is not in the document]
Comment (MaximKYZ) [reply src/types.ts:361]: Done

File: src/redux/slices/flattened-sublayers-slice.tsx [hunk 8]
```diff
@@ -143,7 +143,7 @@ const getLayersAndSublayers = async (
     id: string;
     url: string;
     visibility: boolean;
-    token: string;
+    token: string | undefined;
```
[the remark this answers is not in the document]
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.tsx:146]: Done

File: src/redux/slices/colors-by-attribute-slice.tsx [hunk 9]
```diff
@@ -3,7 +3,7 @@ import { ColorsByAttribute } from "../../types";
 import { RootState } from "../store";
```
[the remark this answers is not in the document]
Comment (MaximKYZ) [reply src/redux/slices/colors-by-attribute-slice.tsx:3]: Done

File: src/redux/slices/colors-by-attribute-slice.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,68 @@
+import { setupStore } from "../store";
+import reducer, {
+  ColorsByAttributeState,
+  selectColorsByAttribute,
+  setColorsByAttrubute,
+} from "./colors-by-attribute-slice";
+
+it("Reducer should return the initial state", () => {
```
[the remark this answers is not in the document]
Comment (MaximKYZ) [reply src/redux/slices/colors-by-attribute-slice.spec.tsx:8]: Done

File: src/redux/slices/drag-mode-slice.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,36 @@
+import { DragMode } from "../../types";
+import { setupStore } from "../store";
+import reducer, {
+  DragModeState,
+  selectDragMode,
+  setDragMode,
+} from "./drag-mode-slice";
+
+it("Reducer should return the initial state", () => {
```
[the remark this answers is not in the document]
Comment (MaximKYZ) [reply src/redux/slices/drag-mode-slice.spec.tsx:9]: Done

File: src/redux/slices/flattened-sublayers-slice.spec.tsx (hunk 1 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.spec.tsx:213]: Done

File: src/redux/slices/flattened-sublayers-slice.spec.tsx (hunk 2 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.spec.tsx:235]: Done

File: src/redux/slices/flattened-sublayers-slice.spec.tsx (hunk 3 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.spec.tsx:257]: Done

File: src/redux/slices/flattened-sublayers-slice.spec.tsx (hunk 4 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.spec.tsx:294]: Done

File: src/redux/slices/flattened-sublayers-slice.spec.tsx (hunk 5 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.spec.tsx:323]: Done

File: src/redux/slices/flattened-sublayers-slice.spec.tsx (hunk 6 above)
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.spec.tsx:352]: Done

File: src/redux/slices/flattened-sublayers-slice.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,1851 @@
...
+        visibility: true,
+        type: TilesetType.I3S,
+      },
+    ],
+    layerCounter: 1,
+    sublayers: [],
+  },
+};
+
+it("Reducer should return the initial state", () => {
```
[the remark this answers is not in the document]
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.spec.tsx:68]: Done

File: src/redux/slices/flattened-sublayers-slice.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,1851 @@
...
+    getFlattenedSublayers({ tilesetsData, buildingExplorerOpened: false })
+  );
+  await sleep(100);
+  const newSingleState = store.getState();
+  expect(selectLayers(newSingleState)).toEqual(wslExpectedLayers);
+  expect(selectSublayers(newSingleState)).toEqual([]);
+  expect(newSingleState.flattenedSublayers.single.layerCounter).toEqual(1);
+});
+
+const bslTestTileset = {
```
[the remark this answers is not in the document]
Comment (MaximKYZ) [reply src/redux/slices/flattened-sublayers-slice.spec.tsx:380]: Done

File: src/redux/slices/uv-debug-texture-slice.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+  const store = setupStore();
+  const state = store.getState();
+  expect(selectUVDebugTexture(state)).toEqual(null);
+
+  store.dispatch(fetchUVDebugTexture());
+  expect(load).toHaveBeenCalledWith(
+    "https://raw.githubusercontent.com/visgl/deck.gl-data/master/images/uv-debug-texture.jpg",
+    ImageLoader
+  );
+  await sleep(100);
```
[the remark this answers is not in the document]
Comment (MaximKYZ) [reply src/redux/slices/uv-debug-texture-slice.spec.tsx:37]: Done

**Review by belom88 — APPROVED** [review]

**Pull request merged**